### ISSUE TITLE
Could you add OnImageAvailableListener into Camera2?

### ISSUE THREAD

**Issue by kikokuo** [pedroSG94/RootEncoder#1166]
Hello,
          I using this project for MLKIT of google. When I using face detect of MLKIT,  the performance not good.
          I try to find solution, but it need to get raw data from camera. So,  Could you add OnImageAvailableListener into Camera2 and Open callback for user? thank you.

**Comment by pedroSG94** [comment]
Hello,

I added this methods for it:
```java
addImageListener(int format, int maxImages, Camera2ApiManager.ImageCallback listener)
removeImageListener()
```

**Comment by kikokuo** [comment]
It's work. thanks.

**Comment by kikokuo** [comment]
> Fixed: [dab3e88](https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/commit/dab3e88717cc8181f065c76fe2a5ab7de160a670) According with documentation could be null if no image available: https://developer.android.com/reference/android/media/ImageReader#acquireLatestImage()

Yes, when it return null, no image available. 
I tested you fix patch, it's working great.  thanks.
And if you have time, you can test performance when assign big than 2 of maxImages.
It help to improve perfromance when you need processing some work at preview image.
thanks again.
